src/test/ui/nll/assign_mutable_fields.rs

```diff
@@ -0,0 +1,34 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// Currently, we permit you to assign to individual fields of a mut
+// var, but we do not permit you to use the complete var afterwards.
+// We hope to fix this at some point.
+//
+// FIXME(#21232)
+
+#![feature(nll)]
```

Comment:
it might be good to promote these tests outside of the `nll/` directory, and let the `compare-mode=nll` encode the differences in behavior between NLL and AST-borrowck. Just a thought.

Reply:
(clearly I didn't care enough about whether this comment was addressed to let it hold up my r+. Still, feel free to incorporate this change if you can get to it before bors merges this.)